funcx_sdk/tests/unit/test_client.py

```diff
@@ -147,3 +148,75 @@ def test_pending_tasks_always_fetched():
         if should_fetch:
             args, _ = fxc.web_client.get_task.call_args
             assert sf == args[0]
+
+
+def test_build_container(mocker, login_manager):
+    mock_data = mocker.Mock()
+    mock_data.data = {"container_id": "123-456"}
+    login_manager.get_funcx_web_client.post = mocker.Mock(return_value=mock_data)
+    fxc = funcx.FuncXClient(do_version_check=False, login_manager=login_manager)
+    spec = ContainerSpec(
+        name="MyContainer",
+        pip=[
+            "matplotlib==3.5.1",
+            "numpy==1.18.5",
+        ],
+        # payload_url="https://raw.githubusercontent.com/funcx-faas/funcx-container-service/dev/README.md",  # noqa E501
+        payload_url="https://github.com/funcx-faas/funcx-container-service.git",
+    )
+
+    container_id = fxc.build_container(spec)
+    assert container_id == "123-456"
+    login_manager.get_funcx_web_client.post.assert_called()
+    calls = login_manager.get_funcx_web_client.post.call_args
+    assert calls[0][0] == "containers/build"
+    assert calls[1] == {
+        "data": {
+            "apt": [],
+            "conda": [],
+            "description": None,
+            "name": "MyContainer",
+            "payload_url": "https://github.com/funcx-faas/funcx-container-service.git",
+            "pip": ["matplotlib==3.5.1", "numpy==1.18.5"],
+        }
+    }
+
+
+def test_container_build_status(mocker, login_manager):
+    class MockData(dict):
+        def __init__(self):
+            self["status"] = "building"
+            self.http_status = 200
+
+    login_manager.get_funcx_web_client.get = mocker.Mock(return_value=MockData())
+    fxc = funcx.FuncXClient(do_version_check=False, login_manager=login_manager)
+    status = fxc.get_container_build_status("123-434")
+    assert status == "building"
```

Comment:
No knowing where we come to on the other discussion (above), I observe that the _current_ code is, on the happy path, basically
```
return r["status"]
```
Which suggests to me that a stronger test would utilize a random value.  Something like:
```python
def test_container_build_status(mocker, login_manager, randomstring):
  expected_status = randomstring()  # specifically generate afresh for every test
  class MockData(dict):
    def __init__(self):
      self["status"] = expected_status
  ...
  assert status == expected_status
```
I argue this idiom is stronger because it reduces the possibility of using a value that "just happens" to work.

Reply:
I like it - added random string to test